custom_components/tahoma/climate_devices/atlantic_heat_recovery_ventilation.py

```diff
@@ -0,0 +1,252 @@
+
+    @callback
+    def update_temp(self, state):
+        """Update thermostat with latest state from sensor."""
+        if state is None or state.state == STATE_UNKNOWN:
+            return
+
+        try:
+            self._current_temperature = float(state.state)
+        except ValueError as ex:
+            _LOGGER.error("Unable to update from sensor: %s", ex)
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return SUPPORT_PRESET_MODE | SUPPORT_FAN_MODE
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return HVAC_MODE_FAN_ONLY
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return the list of available hvac operation modes."""
+        return HVAC_MODE_FAN_ONLY
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., auto, smart, interval, favorite."""
+        state_ventilation_configuration = self.select_state(
+            IO_VENTILATION_CONFIGURATION_MODE_STATE
+        )
+        state_ventilation_mode = self.select_state(IO_VENTILATION_MODE_STATE)
+        state_prog = state_ventilation_mode.get("prog")
+
+        if state_prog == "on":
+            return PRESET_PROG
+
+        if state_ventilation_configuration == "comfort":
+            return PRESET_AUTO
+
+        if state_ventilation_configuration == "standard":
+            return PRESET_MANUAL
+
+        return None
+
+    @property
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return a list of available preset modes."""
+        return PRESET_MODES
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set the preset mode of the fan."""
+        if preset_mode == PRESET_AUTO:
+            await self.async_execute_command(
+                COMMAND_SET_VENTILATION_CONFIGURATION_MODE, "comfort"
+            )
+            await self._set_ventilation_mode(prog="off")
+
+        if preset_mode == PRESET_PROG:
+            await self.async_execute_command(
+                COMMAND_SET_VENTILATION_CONFIGURATION_MODE, "standard"
+            )
+            await self._set_ventilation_mode(prog="on")
+
+        if preset_mode == PRESET_MANUAL:
+            await self.async_execute_command(
+                COMMAND_SET_VENTILATION_CONFIGURATION_MODE, "standard"
+            )
+            await self._set_ventilation_mode(prog="off")
+
+        # TODO Get rid of sleep command
+        asyncio.sleep(0.5)
```

Comment:
@tillstaff is a time between the commands required? They should be posted in a queue and this works in other climate implementations already.

Reply:
@iMicknl : from my tests, the refresh doesn't work if it's run too quickly
I will retest this just in case